tests/utilities/test_auto_restart.py

```diff
@@ -751,10 +755,14 @@ def on_train_batch_start(self, trainer, *_) -> None:
         model = TestModel()
         model.training_epoch_end = None
         trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, limit_train_batches=1, callbacks=Check())
+        trainer._validate_fault_tolerant_training = lambda x, y: None
         trainer.fit(model)
 
 
 class SequentialGetItemDataset(Dataset):
+
+    deterministic = True
```

Comment:
I don't understand this and how it relates to the error in the tests
"Fault Tolerant Training supports only a single dataloader"

Can you explain?

Reply:
Yes, those tests would fail as the dataset are determinitic but Lightning can't infer it. I added this properly, but definitely open to rethink this.